var/spack/repos/builtin/packages/py-nvidia-dali/package.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+
+from spack.package import *
+
+
+class PyNvidiaDali(PythonPackage):
+    """A GPU-accelerated library containing highly optimized building blocks and
+    an execution engine for data processing to accelerate deep learning training
+    and inference applications. """
+
+    homepage = "https://developer.nvidia.com/dali"
+    url = "https://developer.download.nvidia.com/compute/redist/"
+
+    maintainers = ["thomas-bouvier"]
+
+    arch = platform.machine()
+    if arch == "x86_64":
+        version(
+            "1.22.0-cuda120",
+            sha256="6cbd9e3139d4c203f61f960f5ad1fc4b461621a60b7fa7ef0ba6d77c780b35f4",
+            expand=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda120/nvidia_dali_cuda120-1.22.0-6971317-py3-none-manylinux2014_x86_64.whl",
+        )
+        version(
+            "1.22.0-cuda110",
+            sha256="8c3ccc7eddc1f63d3f858448c5c384ab129273e0c140e091aca2a98d48c5a80c",
+            expand=False,
+            preferred=True,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.22.0-6988993-py3-none-manylinux2014_x86_64.whl",
+        )
+    elif arch == "aarch64":
+        version(
+            "1.22.0-cuda120",
+            sha256="5e496eebeba3bc1cddd18e081c8c45121283478931cbe9b64912d2394d0942ca",
+            expand=False,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda120/nvidia_dali_cuda120-1.22.0-6971317-py3-none-manylinux2014_aarch64.whl",
+        )
+        version(
+            "1.22.0-cuda110",
+            sha256="0da47629fec01abf418fda0eeb393998820e40f6fae6b4c7d3e625aa4cdba6bd",
+            expand=False,
+            preferred=True,
+            url="https://developer.download.nvidia.com/compute/redist/nvidia-dali-cuda110/nvidia_dali_cuda110-1.22.0-6988993-py3-none-manylinux2014_aarch64.whl",
+        )
+
+    depends_on("cuda@12", when="@1.22.0-cuda120", type=("build", "run"))
+    depends_on("cuda@11", when="@1.22.0-cuda110", type=("build", "run"))
```

Comment:
Should we use `cuda@12.0` and `@11.0` or are these compatible with all minor releases?

Reply:
Yes, all minor releases are compatible. From the [docs](https://docs.nvidia.com/deeplearning/dali/main-user-guide/docs/installation.html#nvidia-dali) "CUDA 11.0 and CUDA 12.0 build uses CUDA toolkit enhanced compatibility. It is built with the latest CUDA 11.x/12.x respectively toolkit while it can run on the latest, stable CUDA 11.0 and CUDA 12.0 capable drivers (450.80 or later and 525.60 or later respectively). Using the latest driver may enable additional functionality."